worker/jobTypes/productionDeployJob.js

```diff
@@ -0,0 +1,182 @@
+
+const GitHubJob = require('../jobTypes/githubJob').GitHubJobClass;
+const S3Publish = require('../jobTypes/S3Publish').S3PublishClass;
+const simpleGit = require('simple-git/promise');
+const validator = require('validator');
+const workerUtils = require('../utils/utils');
+const buildTimeout = 60 * 450;
+const uploadToS3Timeout = 20;
+
+const invalidJobDef = new Error('job not valid');
+
+async function verifyUserEntitlements(currentJob){
+    const user = currentJob.user;
+    const entitlementsObject = await workerUtils.getUserEntitlements(user);
+    if (entitlementsObject && entitlementsObject.repos && entitlementsObject.repos.indexOf(`${repoOwner}/${repoName}`) !== -1) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+  
+  async function verifyBranchConfiguredForPublish(currentJob) {
+    const repoObject = { repoOwner: currentJob.payload.repoOwner, repoName: currentJob.payload.repoName};
+    const repoContent = await workerUtils.getRepoPublishedBranches(repoObject);
+    const publishedBranches = repoContent['content']['git']['branches']['published']
+
+    if (publishedBranches.includes(currentJob.payload.branchName)) {
```

Comment:
style point -- you could simply return publishedBranches.includes(currentJob.payload.branchName);

Reply:
+1